Neos.ContentRepository.Core/Classes/Feature/Common/ConstraintChecks.php

```diff
@@ -324,37 +327,16 @@ protected function requireNodeTypeConstraintsImposedByParentToBeMet(
                 1707561400
             );
         }
-        if (
-            $nodeName
-            && $parentsNodeType->hasTetheredNode($nodeName)
-            && !$this->getNodeTypeManager()->getTypeOfTetheredNode($parentsNodeType, $nodeName)->name->equals($nodeType->name)
-        ) {
-            throw new NodeConstraintException(
-                'Node type "' . $nodeType->name->value . '" does not match configured "'
-                    . $this->getNodeTypeManager()->getTypeOfTetheredNode($parentsNodeType, $nodeName)->name->value
-                    . '" for auto created child nodes for parent type "' . $parentsNodeType->name->value
-                    . '" with name "' . $nodeName->value . '"',
-                1707561404
-            );
-        }
```

Comment:
why were these constraint checks removed? I came across those many times and last time i read them they seemed correct?

Reply:
They were replaced by an even stricter  requireNodeTypeNotToDeclareTetheredChildNodeName (see above)

Reply:
Okay i see but my feeling says `requireNodeTypeNotToDeclareTetheredChildNodeName` is not called often enough? Only once in the Move handling?

Reply:
It's called three times, at all relevant locations:
* NodeCreation: You can't manually create a missing tethered node anymore
* NodeMove: You can't move a node to a place where a tethered child of the same name is missing
* NodeRenamig: You can't assign the name of a missing tethered sibling to a node any more

Reply:
Wait but what about `handleCopyNodesRecursively`? it had a call previously to `requireConstraintsImposedByAncestorsAreMet` but not to `requireNodeTypeNotToDeclareTetheredChildNodeName`?

Reply:
Good point, I'll have to check what copy does with names.
Type change is unrelated though as it does not affect the name